### ISSUE TITLE
Android 11 - Failed to install

### ISSUE THREAD

**Issue by WimVDK** [topjohnwu/Magisk#3153]
This post is merely for those that forgot to update to canary and had trouble installing android 11 and experienced a boot failure.

**TLDR: Install latest canary manager and change update channel to canary. Then either patch the boot image or (install to inactive slot)**

A11 was just pushed to my pixel 3 XL, i did the standard approach of restoring images, updating and installing to inactive slot and then a magisk reboot. Device restarted a couple times, then booted back to the PREVIOUS BOOT SLOT with magisk no longer installed and still on android 10!

Went back into settings, clicked "try again" and got greeted with an update notification "failed to install". So figured ah well, i'll just go the factory image way and remove the -w flag. Device succesfully updated.

Then i copied the factory image to my device, patched it with magisk 20.4 using the normal settings (preserve force encryption and preserve AVB 2.0). Booted into fastboot and did "fastboot flash boot magisk_patched.img" which succesfully flashes the patched boot image. Followed up with the usual "fastboot reboot". Google logo appears for a few seconds, then boots back into the bootloader. "Enter reason: **NO VALID SLOT TO BOOT**

Simply flashing the stock A11 boot image and the device boots up again normally. Obviously without magisk.

I tried installing the current latest canary but it said it failed to install.  Went into magisk and it was installed anyway :s 

That said, redid the boot image patching (this time with canary), redid boot image flashing and rebooting, **THIS TIME WITH SUCCESS** Using the latest canary available. The device booted right up, **WITH MAGISK INSTALLED**.

So for those of you that wanted to be on A11 on launch day. Make sure you're running the latest canary build, or the OTA will fail to install and the device will either fail to boot or boot back into the previous slot with android 10

**Comment by stanleylhs** [comment]
Wish I have read your post a bit earlier. Bootlooped after OTA. After a few loop, it went back to Android 10.

No need to reflash the factory image as stated in OP, as the new Android 11 in the other slot is still intact just that boot.img is corrupted.

1. Just extract patch the boot image of Android 11 in the Canary build Magisk manager while in Android 10.
1. Then `adb reboot fastboot` 
1. `fastboot --set-active=other` 
1. `fastboot flash boot YOUR_PATCHED_ANDROID11_BOOT`
1. reboot

Viola! Android 11 with Magisk working!

**Comment by WimVDK** [comment]
when a loop happens like this, i err on the side of caution and always flash an image (while editing the flash-all file and remove the -w flag to keep data)

**Comment by prashantwosti** [comment]
> Wish I have read your post a bit earlier. Bootlooped after OTA. After a few loop, it went back to Android 10.
> 
> No need to reflash the factory image as stated in OP, as the new Android 11 in the other slot is still intact just that boot.img is corrupted.
> 
>     1. Just extract patch the boot image of Android 11 in the Canary build Magisk manager while in Android 10.
>        Get canary from here: https://github.com/topjohnwu/magisk_files/tree/canary
> 
>     2. Then `adb reboot fastboot`
> 
>     3. `fastboot --set-active=other`
> 
>     4. `fastboot flash boot YOUR_PATCHED_ANDROID11_BOOT.img`
> 
>     5. reboot
> 
> 
> Viola! Android 11 with Magisk working!

Unfortunately, its not working for me(Pixel 2xl). Tried your process multiple times.

**Comment by prashantwosti** [comment]
Update: you dont have to change the slot. manually patch and flash the boot image and it works.
but safety net is failing.

**Comment by 109021017** [comment]
> > Wish I have read your post a bit earlier. Bootlooped after OTA. After a few loop, it went back to Android 10.
> > No need to reflash the factory image as stated in OP, as the new Android 11 in the other slot is still intact just that boot.img is corrupted.
> > ```
> > 1. Just extract patch the boot image of Android 11 in the Canary build Magisk manager while in Android 10.
> >    Get canary from here: https://github.com/topjohnwu/magisk_files/tree/canary
> > 
> > 2. Then `adb reboot fastboot`
> > 
> > 3. `fastboot --set-active=other`
> > 
> > 4. `fastboot flash boot YOUR_PATCHED_ANDROID11_BOOT.img`
> > 
> > 5. reboot
> > ```
> > 
> > 
> > Viola! Android 11 with Magisk working!
> 
> @stanleylhs Unfortunately, its not working for me(Pixel 2xl, with `taimen-rp1a.200720.009`). Tried your process multiple times. Are you sure about flashing patched boot image to another slot?

Same here, My Pixel 2xl will stuck at Google logo while booting after I flash the magisk_patched.img If I don't change the slot. If I change the slot, it will reboot by itself several times and then boot into system without magisk installed.

**Comment by prashantwosti** [comment]
> > > Wish I have read your post a bit earlier. Bootlooped after OTA. After a few loop, it went back to Android 10.
> > > No need to reflash the factory image as stated in OP, as the new Android 11 in the other slot is still intact just that boot.img is corrupted.
> > > ```
> > > 1. Just extract patch the boot image of Android 11 in the Canary build Magisk manager while in Android 10.
> > >    Get canary from here: https://github.com/topjohnwu/magisk_files/tree/canary
> > > 
> > > 2. Then `adb reboot fastboot`
> > > 
> > > 3. `fastboot --set-active=other`
> > > 
> > > 4. `fastboot flash boot YOUR_PATCHED_ANDROID11_BOOT.img`
> > > 
> > > 5. reboot
> > > ```
> > > 
> > > 
> > > Viola! Android 11 with Magisk working!
> > 
> > @stanleylhs Unfortunately, its not working for me(Pixel 2xl, with `taimen-rp1a.200720.009`). Tried your process multiple times. Are you sure about flashing patched boot image to another slot?
> 
> Same here, My Pixel 2xl will stuck at Google logo while booting after I flash the magisk_patched.img If I don't change the slot. If I change the slot, it will reboot by itself several times and then boot into system without magisk installed. 

I got everything working finally. in order to pass the safetynet use `MagiskHide Props Config`.

**Comment by 109021017** [comment]
@prashantwosti How did you make it work? I have tried patch and flash several times with or without flashing to the other slot without any luck.

**Comment by alexstaj** [comment]
yes pixel2xl here also had difficulty.  is it working now with 20425?

**Comment by prashantwosti** [comment]
> @prashantwosti How did you make it work? I have tried patch and flash several times with or without flashing to the other slot without any luck. 

i flashed the patched boot image using canary magisk 20425 in the same slot. i tried few times back and forth. good luck. let me know if you need any specific details. see my above comments where i have tried to explain. 

![Screenshot_20200911-000907.png](https://user-images.githubusercontent.com/2226096/92742792-51840400-f3c3-11ea-9e23-096a1db3a528.png)

**Comment by WimVDK** [comment]
Make sure you're using the latest canary then patch the boot image or install to inactive slot. I patched the boot image since i was already on A11. Canary should work with the normal OTA.

**Comment by 109021017** [comment]
@prashantwosti Turns out my problem was cause by some moudels I installed that was not compatible with Android 11. So after I delete the /data/adb/modules folder, my device boot up into system successfully with Magisk installed.
